Extensions/Physics2Behavior/tests/physics2runtimebehavior.spec.js

```diff
@@ -190,28 +197,39 @@ describe('Physics2RuntimeBehavior', () => {
           });
         } else {
           hasBounced = true;
-          expect(movingObject.getY() < staticObject.getY()).to.be(true);
-          assertCollision(movingObject, staticObject, {
-            started: true,
-            collision: true,
-            stopped: false,
-          });
+          subStepIndex += 1;
+          if (subStepIndex === 1) {
+            // At first frame, collision should have only started
+            expect(movingObject.getY() < staticObject.getY()).to.be(true);
+            assertCollision(movingObject, staticObject, {
+              started: true,
+              collision: true,
+              stopped: false,
+            });
+          } else if (subStepIndex === 3) {
+            // At next frame, end of collision should be detected
+            assertCollision(movingObject, staticObject, {
+              started: false,
+              collision: false,
+              stopped: true,
+            });
+            hasCollisionEndedBeenTested = true;
+          }
         }
+      });
 
+      while (stepIndex < 10 && !hasBounced) {
+        runtimeScene.renderAndStep(1000 / fps);
         stepIndex++;
       }
+
       if (!hasBounced) {
         throw new Error('Contact did not happen, nothing was tested.');
       }
 
-      runtimeScene.renderAndStep(1000 / fps);
-
-      // At next frame, end of collision should be detected
-      assertCollision(movingObject, staticObject, {
-        started: false,
-        collision: false,
-        stopped: true,
-      });
+      if (!hasCollisionEndedBeenTested) {
+        runtimeScene.renderAndStep(1000 / fps);
+      }
```

Comment:
You should check for the subStepIndex or anything else that will check your test did run through the end (if I comment `renderAndStep`, the test will work).
The `if` also sounds suspicious (in which case can it be false/true? Is it a flaky test?)